### PR TITLE
Is91/review email templates (⚠️ devops)

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

**TL;DR** every product needs custom templates that should be added/updated at any time w/o changing the code-base: here we use the pg database

- ✨ new ``jinja2_templates`` table and a column for email registration in the ``products`` table
- ✨ issue tracker url columns of the product table as now nullable
- ♻️ split ``products`` plugin into multiple modules and tests
- ♻️ cleanup ``templates`` and changed extension to ``.jinja2`` since they are not html but a file with multiple parts (subject line + comment+ html).
- ♻️ cleanup ``login.utils`` and refactor ``common/templates`` to target products 

###  ⚠️ devops

- update ``jinja2_templates`` and ``products`` tables with proper
- create a separate git repository with file copies of these templates 

## Related issue/s

- ITISFoundation/osparc-issues#91

## How to test

```cmd
cd packages/postgres-database
make install-dev
make tests


cd services/web/server
make install-dev
pytest -vv --pdb tests/**/test_product*.py
pytest -vv --pdb tests/**/test_login*.py
```

## Checklist

- [x] Openapi changes? ``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Database migration script? ``cd packages/postgres-database``, ``make setup-commit``, ``sc-pg review -m "my changes"``
- [x] Unit tests for the changes exist
- [x] Runs in the swarm
- [x] Documentation reflects the changes
- [x] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: packages/postgres-database/src/simcore_postgres_database/migration/versions/e5f3167d7277_add_template_col_to_product_table.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        "products", sa.Column("registration_email_template", sa.String(), nullable=True)
+    )
+    op.create_foreign_key(
+        "fk_jinja2_templates_name",
+        "products",
+        "jinja2_templates",
+        ["registration_email_template"],
+        ["name"],
+        onupdate="CASCADE",
+        ondelete="SET NULL",
```
Comment (sanderegg) [inline packages/postgres-database/src/simcore_postgres_database/migration/versions/e5f3167d7277_add_template_col_to_product_table.py:30]: good to know

File: services/web/server/src/simcore_service_webserver/login/_2fa.py [hunk 2]
```diff
@@ -80,6 +80,7 @@ async def send_sms_code(
     twilo_auth: TwilioSettings,
     twilio_messaging_sid: str,
     product_display_name: str,
+    user_name: str = "user",
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/login/_2fa.py:83]: ok, so same question, why do you set a default value although this is a private module?

File: services/web/server/src/simcore_service_webserver/login/_2fa.py [hunk 3]
```diff
@@ -92,8 +93,9 @@ def sender():
         client = Client(twilo_auth.TWILIO_ACCOUNT_SID, twilo_auth.TWILIO_AUTH_TOKEN)
         message = client.messages.create(
             messaging_service_sid=twilio_messaging_sid,
+            from_=product_display_name,
             to=phone_number,
-            body=f"Dear {product_display_name} user, your verification code is {code}",
+            body=f"Dear {user_name[:20].capitalize().strip()}, your verification code is {code}",
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/login/_2fa.py:98]: why 20?

File: services/web/server/src/simcore_service_webserver/login/handlers.py [hunk 4]
```diff
@@ -42,13 +42,19 @@
...
     render_and_send_mail,
-    themed,
 )
 
 log = logging.getLogger(__name__)
 
 
+def _get_user_name(email: str) -> str:
+    username = email.split("@")[0]
+    # TODO: this has to be unique and add this in user registration!
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/login/handlers.py:54]: issue?

**Review by sanderegg — APPROVED** [review]
great!

**Review by odeimaiz — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/login/_2fa.py (hunk 3 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/login/_2fa.py:98]: I wanted to limit the size of the SMS. I looked in the database and the largest name was around 15 characters.

File: services/web/server/src/simcore_service_webserver/login/_2fa.py (hunk 2 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/login/_2fa.py:83]: i would like to listen to your rationale about this point :-)
